### PR TITLE
[SYCL] Remove fabs and ceil from the list of unsupported math functions

### PR DESCRIPTION
**Pull request by asudarsa**
Signed-off-by: Arvind Sudarsanam <arvind.sudarsanam@intel.com>

### CODE REVIEW

**Comment by asudarsa** [comment]
Reviewed by Alexey Sotkin at otcshre/901 PR.
Already submitted to xmain.

File: clang/test/SemaSYCL/unsupported_math.cpp [hunk 1]
```diff
@@ -30,14 +26,6 @@ int main() {
     acc[0] += (int)log(1.0);                    // expected-no-error
     acc[0] += (int)__builtin_logf(1.0f);        // expected-no-error
     acc[0] += (int)__builtin_log(1.0);          // expected-no-error
-    acc[0] += (int)ceilf(1.0f);                 // expected-error{{builtin is not supported on this target}}
```
Comment (bader) [inline clang/test/SemaSYCL/unsupported_math.cpp]:
```suggestion
    acc[0] += (int)ceilf(1.0f);                 // expected-no-error
```

File: clang/lib/Sema/SemaSYCL.cpp [hunk 2]
```diff
@@ -159,11 +155,7 @@ static bool IsSyclMathFunc(unsigned BuiltinID) {
   case Builtin::BItrunc:
   case Builtin::BI__builtin_trunc:
   case Builtin::BIceilf:
-  case Builtin::BI__builtin_ceilf:
-  case Builtin::BIcopysignf:
```
Comment (bader) [inline clang/lib/Sema/SemaSYCL.cpp]: Please, remove line 161 instead of this line.

Comment (asudarsa) [reply clang/lib/Sema/SemaSYCL.cpp]: Submitted a fix. Thanks.

File: clang/test/SemaSYCL/unsupported_math.cpp (hunk 1 above)
Comment (asudarsa) [reply clang/test/SemaSYCL/unsupported_math.cpp]: I think this is overkill as we could end up adding (expected-no-error) for all the supported math library functions.
I agree that my error in the compiler could have been caught by this though.

Comment (bader) [reply clang/test/SemaSYCL/unsupported_math.cpp]: :)

**Pull request merged**